packages/url_launcher/url_launcher_web/lib/src/link.dart

```diff
@@ -197,41 +343,99 @@ class LinkViewController extends PlatformViewController {
     //    - If `_hitTestedViewId` is set, it means the app triggered the link.
     //    - We navigate to the Link's URI.
 
+    if (_isModifierKey(event)) {
+      // Modifier keys (i.e. Shift, Ctrl, Alt, Meta) cannot trigger a Link.
+      return;
+    }
+
     // The keydown event is not directly associated with the target Link, so
-    // we need to look for the recently hit tested Link to handle the event.
-    if (_hitTestedViewId != null) {
-      _instances[_hitTestedViewId]?._onDomKeydown();
+    // we can't find the `viewId` from the event.
+    _triggerSignals.registerDomEvent(viewId: null, mouseEvent: null);
+
+    if (_triggerSignals.isReadyToTrigger) {
+      _triggerLink();
     }
-    // After the keyboard event has been received, clean up the hit test state
-    // so we can start fresh on the next event.
-    unregisterHitTest();
   }
 
   static void _onGlobalClick(html.MouseEvent event) {
-    final int? viewId = getViewIdFromTarget(event);
-    _instances[viewId]?._onDomClick(event);
-    // After the DOM click event has been received, clean up the hit test state
-    // so we can start fresh on the next event.
-    unregisterHitTest();
+    handleGlobalClick(
+      event: event,
+      target: event.target as html.Element?,
+    );
+  }
+
+  /// Global click handler that triggers on the `capture` phase. We use `capture`
+  /// because some events may be consumed and prevent further propagation at the
+  /// target. This may lead to issues (see: https://github.com/flutter/flutter/issues/143164)
+  /// where a followLink was executed but the event never bubbles back up to the
+  /// window (e.g. when button semantics obscure the platform view). We make sure
+  /// to only trigger the link if a hit test was registered and remains valid at
+  /// the time the click handler executes.
+  @visibleForTesting
+  static void handleGlobalClick({
+    required html.MouseEvent event,
+    required html.Element? target,
+  }) {
+    // We only want to handle clicks that land on *our* links, whether that's a
+    // platform view link or a semantics link.
+    final int? viewIdFromTarget = _getViewIdFromLink(target) ??
+        _getViewIdFromSemanticLink(target);
```

Comment:
Does this work if `<flutter-view>` is inside shadow DOM? In embedded case in particular, we don't know how exactly a FlutterView is embedded. It could be inside a web component's shadow DOM. In that case `event.target` caught at `window` level will simply point to the web component. It won't tell you the exact element the mouse event happened on.

Another issue is that this relies on the semantic DOM internals, so it can easily break if the structure changes.

Can we use https://api.flutter.dev/flutter/semantics/SemanticsProperties/onTap.html instead?

Reply:
Regarding `<flutter-view>` being inside shadow DOM, that's a very good point that I hadn't considered. I'll test it and fix it.

Regarding relying on the semantic DOM internals, I couldn't find a better way to implement it. Using `onTap` doesn't work because the children of the `Link` widget typically have their own `onTap` which absorbs the tap event so it never reaches the `Link`'s `onTap`.